protocol/lib/margin/risk.go

```diff
@@ -53,6 +53,19 @@ func (a *Risk) IsLiquidatable() bool {
 	return a.MMR.Sign() > 0 && a.MMR.Cmp(a.NC) > 0
 }
 
+// Cmp compares the risks of two accounts.
+// Returns -1 if a is less risky than b, 0 if they are equally risky, and 1 if a is more risky than b.
+
+// Note that here we are effectively checking that
+// `a.NetCollateral / a.MaintenanceMargin >= b.NetCollateral / b.MaintenanceMargin`.
+// However, to avoid rounding errors, we factor this as
+// `a.NetCollateral * b.MaintenanceMargin >= b.NetCollateral * a.MaintenanceMargin`.
+func (a *Risk) Cmp(b Risk) int {
```

Comment:
- consider if `Cmp` is the clearest name as opposed to `CmpRisk` or `CmpSafety` (would be the inverse)
- also consider if we need strict ordering at all (?) like if 2/1 should `Cmp==0` to 4/2

Reply:
Think `Cmp` is fine if we want to make it a method on `Risk`. CmpRisk / CmpSafety sounds like a library function that takes in two `Risk` objects. Let me know if you have a preference here. 

> also consider if we need strict ordering at all (?) like if 2/1 should Cmp==0 to 4/2
Added strict ordering. Tie break using NC if a.MMR = b.MMR = 0, otherwise, tie break using MMR. Added corresponding unit tests 🙏